launcher/agent/agent.go

```diff
@@ -75,16 +95,46 @@ func CreateAttestationAgent(tpm io.ReadWriteCloser, akFetcher util.TpmKeyFetcher
 	if err != nil {
 		return nil, fmt.Errorf("failed to create an Attestation Agent: %w", err)
 	}
-	return &agent{
-		tpm:              tpm,
+
+	attestAgent := &agent{
 		client:           verifierClient,
 		fetchedAK:        ak,
 		principalFetcher: principalFetcher,
 		sigsFetcher:      sigsFetcher,
 		launchSpec:       launchSpec,
 		logger:           logger,
 		sigsCache:        &sigsCache{},
-	}, nil
+	}
+
+	qp, err := tg.GetQuoteProvider()
+	if err != nil {
+		return nil, fmt.Errorf("GetQuoteProvider should always return nil error, but got %v", err)
+	}
+
+	if err := qp.IsSupported(); err != nil {
+		// check if the quote provider is backed by TDX
+		logger.Info(fmt.Sprintf("Cannot create the TDX quote provider (this is expected on a non-TDX machine): %v", err))
+		logger.Info("Using TPM PCRs for measurement")
+		// if cannot get a TDX Quote Provider, by default using TPM as the attest root
+		attestAgent.ar = &tpmAttestRoot{
+			fetchedAK: ak,
+			tpm:       tpm,
+		}
+	} else {
+		logger.Info("Using TDX RTMRs for measurement (Public Preview)")
```

Comment:
remove the Public Preview bit.
"Using TDX RTMRs for measurement"

Reply:
removed